lib/features/popup-menu/ReplaceMenuProvider.js

```diff
@@ -519,6 +523,38 @@ ReplaceMenuProvider.prototype._getDataObjectIsCollection = function(element) {
   return dataObjectEntries;
 };
 
+/**
+ * Get a list of menu items containing a button for the participant multiplicity marker
+ *
+ * @param  {djs.model.Base} element
+ *
+ * @return {Array<Object>} a list of menu items
+ */
+ReplaceMenuProvider.prototype._getParticipantMultiplicity = function(element) {
+
+  var self = this;
+  var translate = this._translate;
+
+  function toggleParticipantMultiplicity(event, entry) {
+    self._modeling.updateProperties(
+      element,
+      { participantMultiplicity: !entry.active });
```

Comment:
I think we have to be careful with this. [According to the spec, this is **not** a boolean](https://github.com/bpmn-io/bpmn-moddle/blob/master/resources/bpmn/json/bpmn.json#L1970), but an element with type `bpmn: ParticipantMultiplicity`. If we store it is as a boolean, we will get in trouble while serializing (`modeler#saveXML`). 

E.g. when using the inside Camunda Modeler, I got the following error when switching to the XML Editor.

```text
/Users/niklas.kiefer/Desktop/diagram_1.bpmn] Cannot read property 'isGeneric' of undefined
    at ElementSerializer.build (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1226:37)
    at eval (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1472:30)
    at forEach (webpack-internal:///../../bpmn-js/node_modules/min-dash/dist/index.esm.js:181:16)
    at eval (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1460:63)
    at forEach (webpack-internal:///../../bpmn-js/node_modules/min-dash/dist/index.esm.js:181:16)
    at ElementSerializer.parseContainments (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1432:59)
    at ElementSerializer.build (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1247:10)
    at eval (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1472:30)
    at forEach (webpack-internal:///../../bpmn-js/node_modules/min-dash/dist/index.esm.js:181:16)
    at eval (webpack-internal:///../../bpmn-js/node_modules/moddle-xml/dist/index.esm.js:1460:63) [ error ]
```

That's because `moddle` does not know what due to the Boolean value.

Reply:
I think better would be something like

```js
var isActive = entry.active;

var participantMultiplicity;

if(isActive) {
  participantMultiplicity = bpmnFactory.create('bpmn:ParticipantMultiplicity');
}

self._modeling.updateProperties(
  element,
  { participantMultiplicity: participantMultiplicity });
);

```

Reply:
[The `BpmnRenderer`](https://github.com/bpmn-io/bpmn-js/blob/5d339676f2e6fb3b095185631d76c83cdaebca95/lib/draw/BpmnRenderer.js#L1170) is also not checking for a boolean only, but for an _existing_ object.

Reply:
Also, cf. this example diagram: https://github.com/bpmn-io/bpmn-js/blob/master/test/fixtures/bpmn/draw/pools-with-collection-marker.bpmn#L4

Reply:
Thanks for the catch, I was a bit too quick to implement it.
I'll look into it and update my PR :+1:

Reply:
🚀 🚀